ext/pcre/php_pcre.c

```diff
@@ -138,6 +127,28 @@ static void pcre_handle_exec_error(int pcre_code) /* {{{ */
 }
 /* }}} */
 
+static const char *php_pcre_get_error_msg(php_pcre_error_code error_code) /* {{{ */
+{
+    switch (error_code) {
+        case PHP_PCRE_INTERNAL_ERROR:
+            return "Internal error";
+        case PHP_PCRE_BAD_UTF8_ERROR:
+        case PHP_PCRE_BAD_UTF8_OFFSET_ERROR:
+            return "Malformed UTF-8 characters, possibly incorrectly encoded";
```

Comment:
Any reason to combine these two errors?

Reply:
Well two, actually. While googling these constants I came across [this gist](https://gist.github.com/cmb69/a9c65af3a1462ed21b12beace502de0d) from @cmb69 and I saw he used the same error message in both cases as well (although this didn't end up in master)

And besides that, it's not really clear to me what the `PHP_PCRE_BAD_UTF8_OFFSET_ERROR` error means, and how to trigger it. Any ideas?

I've just changed the message slightly to distinguish them better.

Reply:
The error is triggered if the string is correctly encoded, but the `$offset` parameter does not point to a character boundary. That is, `$offset` must be at a UTF-8 continuation byte.

Reply:
Okay, understood, thanks! I've added a test case for this scenario.